editor/src/components/inspector/flex-section.tsx

```diff
@@ -1,64 +1,153 @@
 import React from 'react'
 import { createSelector } from 'reselect'
 import { when } from '../../utils/react-conditionals'
-import { Substores, useEditorState } from '../editor/store/store-hook'
+import { Substores, useEditorState, useRefEditorState } from '../editor/store/store-hook'
 import { AddRemoveLayoutSystemControl } from './add-remove-layout-system-control'
 import { FlexDirectionToggle } from './flex-direction-control'
 import { selectedViewsSelector, metadataSelector } from './inpector-selectors'
-import { detectAreElementsFlexContainers } from './inspector-common'
 import { NineBlockControl } from './nine-block-controls'
 import { UIGridRow } from './widgets/ui-grid-row'
-import {
-  DisabledFlexGroupPicker,
-  PaddingRow,
-} from '../../components/inspector/sections/layout-section/layout-system-subsection/layout-system-controls'
+import { LayoutSystemControl } from '../../components/inspector/sections/layout-section/layout-system-subsection/layout-system-controls'
 import { SpacedPackedControl } from './spaced-packed-control'
 import { ThreeBarControl } from './three-bar-control'
 import { FlexGapControl } from './sections/layout-section/flex-container-subsection/flex-container-controls'
 import { FlexContainerControls } from './sections/layout-section/flex-container-subsection/flex-container-subsection'
 import { FlexCol } from 'utopia-api'
+import { MetadataUtils } from '../../core/model/element-metadata-utils'
+import { strictEvery } from '../../core/shared/array-utils'
+import { useDispatch } from '../editor/store/dispatch-context'
+import {
+  addFlexLayoutStrategies,
+  addGridLayoutStrategies,
+} from './inspector-strategies/inspector-strategies'
+import { executeFirstApplicableStrategy } from './inspector-strategies/inspector-strategy'
+import type { DetectedLayoutSystem } from 'utopia-shared/src/types'
+import { assertNever } from '../../core/shared/utils'
+import { Subdued } from '../../uuiui'
 
-export const areElementsFlexContainersSelector = createSelector(
+export const layoutSystemSelector = createSelector(
   metadataSelector,
   selectedViewsSelector,
-  detectAreElementsFlexContainers,
+  (metadata, selectedViews) => {
+    const detectedLayoutSystems = selectedViews.map(
+      (path) =>
+        MetadataUtils.findElementByElementPath(metadata, path)?.specialSizeMeasurements
+          .layoutSystemForChildren ?? null,
```

Comment:
Should this return `'none'` rather than `null`? I'm genuinely unsure

Reply:
Completely fair question, I added the `?? null` to remove the `| undefined` from the type of `specialSizeMeasurements.layoutSystemForChildren` to make it a less noisy type. I didn't add `'none'` to make the type of `detectedLayoutSystems` similar to what we'd get if we checked `specialSizeMeasurements.layoutSystemForChildren` for a single element